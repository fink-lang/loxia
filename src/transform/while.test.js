import {fink2js} from '../testing';


test('while', ()=> {
  expect(
    fink2js(`
      while item:
        item < 10
    `)
  ).toMatchSnapshot();
});
