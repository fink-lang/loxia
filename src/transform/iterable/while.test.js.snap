// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`while 1`] = `
"(function* filter_while(ˆitems_2) {
  for (const ˆitem_1 of ˆitems_2) {
    const item = ˆitem_1;
    const ˆresult_3 = item < 10;
    if (ˆresult_3 !== true) return;
    yield ˆitem_1;
  }
});

Object.assign(module.exports, {});"
`;
